### DISCUSSION TITLE
ATXD35A5V with S403 System is offline. Any solutions?

### DISCUSSION THREAD

**Discussion by RaW79** [revk/xESP32-Faikout#522] [Q&A] [answered]
Hi
Just installed a Faikin at a Siesta Essence (ATXD35A5V1B) connect at known S403 (4 wires excluding 5V). But now the Faikin (current Layout from August 2024) only says "System offline". Yes it's not listed in the supported Models list, but any solutions known?
My Board:
![PXL_20241011_164053151~2](https://github.com/user-attachments/assets/9f8e2bdc-66d0-4362-b391-6a38d613da52)

The LED ist Mangenta. Do I have to Set the model Code manually or do i have to do a Special Setup (inverted TX or RX)?

S403 is connected this way (do not check cabling based on colour, there is a crossing/soldering point in the middle):

![ek-yams-a9651a6e02694b5a9336678d5463dd8c-attachment-0-1728577779642 (002)](https://github.com/user-attachments/assets/fb8b6858-2c11-4042-bc97-4d050031a821)
![grafik](https://github.com/user-attachments/assets/8528af9d-3469-4ff6-b152-d511e9eeb438)

Big thanks!

**Comment by revk** [comment]
It will try inverting tx/rx automatically and try S21 and X50A. So no, it should just work.

Double check what others have posted re the S403.

**Reply by RaW79** [reply]
Thanks this gives me confidence but still don´t know why. S21 & X50A is active in the settings.

Is there any hint to gain by a debug message (log or message over mqtt?) or just by checking the voltage levels at the faikin? read that tx/RX should also be 5V?

Is it possible to set the model manually in the faikin to help the auto detection? Have not found sth. about the possible/correct writing. Is there a list?

**Reply by revk** [reply]
I think S403 uses S21 protocol, you can turn off X50A (the `nox50a` setting). The logs over MQTT may show if some data seen at all. You can also tell it not to try inverting tx and rx.

**Reply by revk** [reply]
The voltage on tx and rx should, IIRC, be 5V when idle. I.e. pulled up. If you have a 5V supply anywhere on the board then connecting that may help as we have an internal pull up as well.

AFAIK the S403 is mains voltages, so be careful!

**Reply by RaW79** [reply]
Thanks. Just keep on trying. deactivated the X50A protocol and also tried with no RX & no TX invertation. I can see no difference.

The MQTT topic "error" just seems to give an TX timeout error:
![grafik](https://github.com/user-attachments/assets/0c13b9ce-19f5-4d35-b45d-7367db520dec)

I checked voltage levels with a simple voltagemeter:

12V Pin:  14.5VDC
RX PIN:   4.95VDC
TX PIN:   4.95VDC
5V PIN: Not connected

But RX & TX goes down to 0V after some seconds of measuring. Because of the high resistance of a voltagemeter I guess this is not because of a load problem but is this ok/normal?

**Reply by revk** [reply]
When connected to aircon the aircon should have internal 5V pull ups. It does in an S21.

If not then you need to find and connect 5V pin to failin to force the pull ups.

**Reply by RaW79** [reply]
Do you know If this is Just for Initial detection at Startup? In an Analysis Here for the S403 someone used 20k Resistor from the 14VDC to the 5VDC Pin of the Faikin. Good Idea for Testing?
If this works would think of usiing a LDO for this instead of doing DIY things on the Daikin PCB. What do you think?

**Reply by revk** [reply]
The 5V pull up is needed fort tx to work as the processor is 3.3V. An S21 normally has one in the air-con.

**Reply by RaW79** [reply, marked as answer]
Thanks for your support and help. It´s fully working without 5V connection or any additional Pullup or so.
First part of the problem was a RX/TX switch in my cable which I recognized very early but solving this issue by a changed configuration in the Faikin setup did not work for me.
So now I changed the wires according standard GPIO usage and this works!
Maybe there is a problem in the faikin software by using GPIO 34 as TX and GPIO 48 as RX, don´t know and better avoid!

**Reply by revk** [reply]
You can't swap the ports. They have hardware that is directional. Did not know you had done that. Glad you sorted.

**Reply by RaW79** [reply]
Thanks and may the config options may offer too much freedom which isn´t really there. Maybe a hint text would be great to improve:
![grafik](https://github.com/user-attachments/assets/7f6c3b89-f9dc-4def-bc90-50e910dd944e)
Nevertheless again staying close to standard isn´t a bad option :)

**Reply by revk** [reply]
The software can run on other boards where the config needs to be configurable. Sorry.

**Reply by RaW79** [reply]
No need for a "sorry". Just mentioning my impression with the intention to help to improve your great solution. A hint in the docs or hint in the menu ("In case of standard faikin module, do not change") would maybe help also others...
